File: LivrusVirtus/FoxToolKit/FXPython/Livrus.py
#! /usr/bin/env python

from FXPy.fox import *
import sys
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
from resource import Resource
import Splash
import Login
import Principal

def main(): 
  LivrusApp = FXApp("Livrus", Resource.STR_TITULO)

  RotinasGlobais.Rotinas = RotinasGlobais()
  ConsultasSQL.ConsSQL = ConsultasSQL()
  Principal.fPrincipal = Principal.create(LivrusApp)
  Login.fLogin = Login.create(Principal.fPrincipal)
  Splash.fSplash = Splash.create(Principal.fPrincipal)

  LivrusApp.init(sys.argv)
  LivrusApp.create()

  if not RotinasGlobais.Rotinas.ConectaBanco("ACLivrus","",""):
    return False
  else:
    #Splash.fSplash.iniciaTimer()
    #Splash.fSplash.show(PLACEMENT_SCREEN)
    Principal.fPrincipal.show(PLACEMENT_SCREEN)

  LivrusApp.run()

if __name__ == '__main__':
  main()

File: LivrusVirtus/VCPP6/FoxToolKit/ODBC/Livrus.cpp
#include "resource.h"
#include "Livrus.h"

int main(int argc,char *argv[]){

  FXApp CLivrusApp("Livrus", STR_TITULO);
  CLivrusApp.init(argc,argv); 

  Rotinas = new CRotinasGlobais(&CLivrusApp);
  frmPrincipal = new CfrmPrincipal(&CLivrusApp);
  frmLogin = new CfrmLogin(&CLivrusApp); 
  frmSplash = new CfrmSplash(&CLivrusApp);

  CLivrusApp.create();

  if (!Rotinas->ConectaBanco("AClivrus","","")) {
	  return FALSE;
  } else {
   frmSplash->iniciaTimer();
   frmSplash->show(PLACEMENT_SCREEN);
  }
  
  return CLivrusApp.run();
}
